template.yaml

```diff
@@ -34,17 +48,59 @@ Resources:
               - sts:AssumeRole
       ManagedPolicyArns:
         - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+        - !Ref BudgetMakerFunctionPolicy
+
+  BudgetMakerFunctionPolicy:
+    Type: AWS::IAM::ManagedPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: BudgetReadWrite
+            Effect: 'Allow'
+            Action:
+              - budgets:ViewBudget
+              - budgets:ModifyBudget
+            Resource: '*'
+
+  BudgetMakerNotificationTopic:
+    Type: AWS::SNS::Topic
+
+  BudgetMakerNotificationTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    Properties:
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+        - Sid: BudgetNotificationPublication
+          Effect: Allow
+          Principal:
+            Service: 'budgets.amazonaws.com'
+          Action: sns:Publish
+          Resource: '*'
+      Topics:
+        - !Ref BudgetMakerNotificationTopic
 
 Outputs:
-  # ServerlessRestApi is an implicit API created out of Events key under Serverless::Function
-  # Find out more about other implicit resources you can reference within SAM
-  # https://github.com/awslabs/serverless-application-model/blob/master/docs/internals/generated_resources.rst#api
-  # HelloWorldApi:
-  #   Description: "API Gateway endpoint URL for Prod stage for Hello World function"
-  #   Value: !Sub "https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/hello/"
   BudgetFunctionArn:
-    Description: "Budget-making Lambda Function ARN"
+    Description: 'Budget-making Lambda Function ARN'
     Value: !GetAtt BudgetMakerFunction.Arn
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-FunctionArn'
   BudgetFunctionRoleArn:
-    Description: "IAM Role created for Budget-making function"
+    Description: 'IAM Role created for Budget-making function'
     Value: !GetAtt BudgetMakerFunctionRole.Arn
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-RoleArn'
+  BudgetMakerFunctionPolicyArn:
+    Value: !Ref BudgetMakerFunctionPolicy
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-FunctionPolicyArn'
+  BudgetMakerNotificationTopicArn:
+    Value: !Ref 'BudgetMakerNotificationTopic'
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-SnsTopicArn'
```

Comment:
why not give these exports the same name as the resource name? I think it makes it easier to maintain.  `'${AWS::Region}-${AWS::StackName}-BudgetMakerNotificationTopicArn`

Reply:
In other PRs you have asked me to truncate the name. Let's come up with a standard and stick to it.

Reply:
really?  i always do it the way i suggested

Reply:
The problem of export names was discussed in a video chat and we agreed that the standard will always be to use the actual resource name in the final segment, even if it is repetitive with the stack name.